Sources/Data/models/WalletUserAssetGroup.swift

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+import Shared
+import os.log
+
+public final class WalletUserAssetGroup: NSManagedObject, CRUD {
```

Comment:
An asset group is a group to collect all user assets that have the same coin type and chainId. It will be used to filter user assets more efficiently.